Mathlib/LinearAlgebra/RootSystem/Defs.lean

```diff
@@ -385,6 +396,26 @@ lemma isReduced_iff : P.IsReduced ↔ ∀ i j : ι, i ≠ j →
     · exact Or.inl (congrArg P.root h')
     · exact Or.inr (h i j h' hLin)
 
+variable {P} in
+lemma smul_coroot_eq_of_root_eq_smul [Finite ι] [NoZeroSMulDivisors ℤ N] (i j : ι) (t : R)
+    (h : P.root j = t • P.root i) :
+    t • P.coroot j = P.coroot i := by
+  have hij : t * P.pairing i j = 2 := by simpa using ((P.coroot' j).congr_arg h).symm
+  refine Module.eq_of_mapsTo_reflection_of_mem (f := P.root' i) (g := P.root' i)
+    (finite_range P.coroot) (by simp [hij]) (by simp) (by simp [hij]) (by simp) ?_
+    (P.mapsTo_coreflection_coroot i) (mem_range_self i)
+  convert P.mapsTo_coreflection_coroot j
+  ext x
+  replace h : P.root' j = t • P.root' i := by ext; simp [h, root']
+  simp [Module.preReflection_apply, coreflection_apply, h, smul_comm _ t, mul_smul]
+
+variable {P} in
+lemma root_eq_smul_root_iff [Finite ι] [NoZeroSMulDivisors ℤ M] [NoZeroSMulDivisors ℤ N]
+    {i j : ι} {t : R} :
+    P.root j = t • P.root i ↔ P.coroot i = t • P.coroot j :=
```

Comment:
Arguably it would be handy to spell this as
```suggestion
    P.root j = t • P.root i ↔ t • P.coroot i = P.coroot j :=
```
which can be passed to `simp` without fear of looping

Reply:
I think it can already be passed to `simp` because the LHS has roots and the RHS coroots.

Should the version with coroots on the LHS be a simp lemma?

Reply:
OK flipped sides and made `simp`!